src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -117,6 +122,31 @@ internal class WorkloadInformationProvider : IWorkloadInformationProvider
             }
         }
 
+        public List<int> GetPortsToIgnoreFromAnnotations(V1ServiceList serviceNameSpace) {
+            var portsToIgnore = new List<int>();
+
+            List<V1Service> portsList = serviceNameSpace.Items.Where(item => item.Metadata != null && 
+            item.Metadata.Annotations != null &&
+            item.Metadata.Annotations.ContainsKey(ServiceAnnotations)).ToList();
+
+            portsList.ForEach(port => {
+                if (port.Metadata.Annotations.TryGetValue(ServiceAnnotations, out string ports)) {

```

Comment:
I think we should probably add null check on Metadata and Annotations here:
if (port.Metadata?.Annotations?.TryGetValue)